arch/arm/core/aarch32/isr_wrapper.S

```diff
@@ -134,12 +134,15 @@ _idle_state_cleared:
 #elif defined(CONFIG_ARMV7_M_ARMV8_M_MAINLINE)
 	sub r0, r0, #16	/* get IRQ number */
 	lsl r0, r0, #3	/* table is 8-byte wide */
-#elif defined(CONFIG_ARMV7_R)
-	/*
-	 * Cortex-R only has one IRQ line so the main handler will be at
-	 * offset 0 of the table.
-	 */
-	mov r0, #0
+#elif defined(CONFIG_CPU_CORTEX_A) || defined(CONFIG_CPU_CORTEX_R)
```

Comment:
I don't like that we now mix ARM_ architecture and CORTEX_ version Kconfigs

Reply:
We have had a similar discussion earlier in another PR; the main problem is that there is no "umbrella symbol" for ARMvX-R besides `CONFIG_CPU_CORTEX_R`, and it is foreseeable that this will apply to all ARM R (and A) profile cores.

Should define `ARM_R` symbol for the sake of consistency? (though that would mean exactly the same thing as `CPU_CORTEX_R`, unless ARM decides that they want to have an ARM core that is not "Cortex").

Reply:
Can you then just make it clearer in the lines above?
E.g. enclose all content between lines 127-136 in CONFIG_CPU_CORTEX_M

Reply:
Done